onnxruntime/core/providers/nnapi/nnapi_builtin/builders/impl/split_op_builder.cc

```diff
@@ -0,0 +1,162 @@
+
+  // Split opset 13- uses "split" as attribute. Currently it's not supported.
+  int GetMinSupportedOpSet(const NodeUnit& /* node_unit */) const override { return 13; }
+
+  // NNAPI Split is available since NNAPI feature level 3
+  int32_t GetMinSupportedNNAPIFeatureLevel(const NodeUnit& /* node_unit */,
+                                           const OpSupportCheckParams& /* params */) const override {
+    return ANEURALNETWORKS_FEATURE_LEVEL_3;
+  }
+};
+
+// Add operator related
+
+void SplitOpBuilder::AddInitializersToSkip(ModelBuilder& model_builder, const NodeUnit& node_unit) const {
+  const auto& input_defs = node_unit.Inputs();
+
+  if (input_defs.size() > 1 && input_defs[1].node_arg.Exists()) {  // optional second input "split"
+    model_builder.AddInitializerToSkip(input_defs[1].node_arg.Name());
+  }
+}
+
+Status SplitOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const NodeUnit& node_unit) const {
+  const auto& input = node_unit.Inputs()[0].node_arg.Name();
+  const auto& output = node_unit.Outputs();
+
+  NodeAttrHelper helper(node_unit);
+  const auto axis = helper.Get("axis", 0);
+
+  int32_t num_outputs;
+  if (node_unit.SinceVersion() >= 18) {
+    num_outputs = SafeInt<int32_t>(helper.GetInt("num_outputs").value());
+  } else {
+    num_outputs = SafeInt<int32_t>(node_unit.Outputs().size());
+  }
+
+  std::vector<std::string> outputs;
+  outputs.reserve(num_outputs);
+  for (int32_t i = 0; i < num_outputs; ++i) {
+    outputs.push_back(output[i].node_arg.Name());
+  }
+
+  ORT_RETURN_IF_ERROR(op_builder_helpers::AddNnapiSplit(model_builder, input, axis, outputs));
+
+  return Status::OK();
+}
+
+// Operator support related
+
+bool SplitOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers, const NodeUnit& node_unit,
+                                       const OpSupportCheckParams& /* params */) const {
+  Shape input_shape;
+  if (!GetShape(node_unit.Inputs()[0].node_arg, input_shape))
+    return false;
+
+  const auto& input_defs = node_unit.Inputs();
+  NodeAttrHelper helper(node_unit);
+  const auto axis = helper.Get("axis", 0);
+
+  const auto split_dims_at_axis = input_shape[HandleNegativeAxis(axis, input_shape.size())];
+  if (input_defs.size() > 1 && input_defs[1].node_arg.Exists()) {
+    // if optional input `split` is provided
+    if (!Contains(initializers, input_defs[1].node_arg.Name())) {
+      LOGS_DEFAULT(VERBOSE) << "New shape of reshape must be known";
+      return false;
+    }
+    const auto& splits_tensor = *initializers.at(input_defs[1].node_arg.Name());
+    Initializer unpacked_tensor(splits_tensor);
+    auto splits_span = unpacked_tensor.DataAsSpan<int64_t>();
+    uint32_t sum_of_splits = SafeInt<uint32_t>(std::accumulate(splits_span.begin(), splits_span.end(), 0));
+    if (sum_of_splits != split_dims_at_axis) {
+      LOGS_DEFAULT(VERBOSE) << "Mismatch between the sum of 'split'. Expected: "
```

Comment:
nit: a mismatch is between at least two things. also can be clearer about what the expected and actual quantities are.